src/distance.cpp

```diff
@@ -0,0 +1,81 @@
+#include "attitude-utils.hpp"
+#include "style.hpp"
+#include "camera.hpp"
+#include "distance.hpp"
+#include <math.h>
+
+namespace found {
+    class SphericalDistanceDeterminationAlgorithm {
+    public:         
+        distFromEarth Run(char *image, Points &p, int imageWidth, int imageHeight)
+        {
+            return solve(p, radius_);
+        }
+    private:
+
+        Vec3 getCenter(Vec3* spats){
+            Vec3 diff1 = spats[1] - spats[0];
+            Vec3 diff2 = spats[2] -spats[1];
+
+            Vec3 circleN = diff1.CrossProduct(diff2);
+            Vec3 circlePt = spats[0];
+
+            Vec3 mid1 = midpoint(spats[0], spats[1]);
+            Vec3 mid2 = midpoint(spats[1], spats[2]);
+
+            Vec3 mid1N = diff1;
+            Vec3 mid2N = diff2;
+
+            Mat3 matrix;
+            matrix.x[0] = circleN.x;
+            matrix.x[1] = circleN.y;
+            matrix.x[2] = circleN.z;
+            matrix.x[3] = mid1N.x;
+            matrix.x[4] = mid1N.y;
+            matrix.x[5] = mid1N.z;
+            matrix.x[6] = mid2N.x;
+            matrix.x[7] = mid2N.y;
+            matrix.x[8] = mid2N.z; 
+
+            int alpha = circleN*circlePt;
+            int beta = mid1N*mid1;
+            int gamma = mid2N*mid2;
+
+            Vec3 y;
+            y.x = alpha;
+            y.y = beta;
+            y.z = gamma;
+
+            Vec3 center = y * matrix.Inverse();
+
+            return center;
+        }
+
+        decimal getRadius(Vec3* spats, Vec3 center){
+            return Distance(spats[0], center);
+        }
+
+        decimal getDistance(decimal r)
+        {
+            return 6378/r;
```

Comment:
Use a macro to define this "magic number" so that know what it means.

The define macro basically defines a "constant" in your system.
`#define RADIUS_OF_EARTH 6378.0`

Technically, Google Style Guides don't like this, so if you want to do something else, then use the following:
`static constexpr decimal radius_of_earth = 6378.0`